civihr_employee_portal/src/View/CustomFieldsMappingFixer.php

```diff
@@ -0,0 +1,164 @@
+ * mechanisms thus producing different IDs.
+ */
+class CustomFieldsMappingFixer {
+
+  /**
+   * To fix mapping on Views import of CiviCRM custom fields
+   *
+   * @param view $view
+   *     the view that is being imported / recreated
+   */
+  public static function fixMapping($view) {
+    foreach ($view->display as $display) {
+      foreach ($display->display_options as &$displayOption) {
+        if (!is_array($displayOption)) {
+          continue;
+        }
+
+        self::processDisplayOptions($displayOption);
+      }
+    }
+  }
+
+  /**
+   * Process all display options for an exported view only when it is related
+   * to a field and the field has a number at the end of its name, which is
+   * the pattern for custom fields from CiviCRM
+   *
+   * @var array
+   *     Display option a exported view structure
+   */
+  private static function processDisplayOptions(&$displayOption) {
+    foreach ($displayOption as &$option) {
+      // only continue if this option is about a field
+      if (!isset($option['field'])) {
+        continue;
+      }
+
+      $optionIdParts = explode('_', $option['field']);
+      $finalIdPart = array_pop($optionIdParts);
+      if (count($optionIdParts) === 0 || !intval($finalIdPart)) {
+        continue;
+      }
+
+      self::processOption($option);
+    }
+  }
+
+  /**
+   * To process a display option for a field in view
+   *
+   * @var array $option
+   *     structure that contains an option for a field in view
+   */
+  private static function processOption(&$option) {
+    // checking if this table is part of the civicrm views integration
+    $tableName = self::getCiviCRMtableName($option['table']);
+    $prefix = $tableName ? self::getPrefix($tableName) : false;
+    if ( $prefix ) {
+      $option['table'] = $tableName;
+      $columns = self::getColumnsFromTable($prefix . $tableName);
+      // if the table where the current field belongs exists does not have it
+      // then we can search for similar fields in the same table and make replacements
+      if (!isset($columns[$option['field']])) {
+        $fieldNameBase = self::removeLastNumberOfString($option['field']);
+        $pos = array_search($fieldNameBase, $columns);
+        if ($pos !== FALSE) {
+          $option['field'] = array_search($columns[$pos], $columns);
+        }
+      }
+    }
+  }
+
+  /**
+   * Removes the number in end of a string
+   * In this context that number is always separated by an underscore
+   *
+   * @param  string $string
+   *     the string which could have underscores or not
+   *
+   * @return string
+   *     the string without the number at the end,
+   *     if no number found at the end returns the same string
+   */
+  private static function removeLastNumberOfString($string) {
+    $pieces = explode('_', $string);
+    $lastIndex = count($pieces) - 1;
+    return $lastIndex > 0 && intval(array_pop($pieces)) ? implode('_', $pieces) : $string;
```

Comment:
space before `return`, also this long line of logic would be a lot more readable with separate variables like

```
$pieces = explode('_', $string);
$lastIndex = count($pieces) - 1;
$hasEnoughParts === ($lastIndex > 0);
if (!$hasEnoughParts) {
  return $string;
}
$lastPart = array_pop($pieces);
$lastPartIsInt === is_int($lastPart);

return  $lastPartIsInt ? implode('_', $pieces) : $string;
```

Although I think if I was writing this I'd probably avoid expensive `explode` and `implode` calls and just use some function that used `strrpos` and `substr`